### PR TITLE
chore: bump android-sdk-ui from 23.3.0 to 24.3.0 and replace all com.appboy imports with com.braze

### PR DESCRIPTION
**Pull request by techouse**
## Summary
Fixes https://github.com/mparticle-integrations/mparticle-android-integration-appboy/pull/115, https://github.com/mparticle-integrations/mparticle-android-integration-appboy/pull/128 and addresses https://github.com/mparticle-integrations/mparticle-android-integration-appboy/issues/117

The Braze Android SDK v24 replaced the `com.appboy` classpath with `com.braze` as [outlined in the changelog](https://github.com/Appboy/appboy-android-sdk/commit/443b232c545d5127bca0cd36001b5b421385a119).

Currently, this mParticle kit still looks for `com.appboy` which crashes the app. This PR fixes it by replacing all `com.appboy` imports with their `com.braze` equivalents.

|old import|replaced by|
|---|---|
|`com.appboy.enums.Gender`|`com.braze.enums.Gender`|
|`com.appboy.enums.Month`|`com.braze.enums.Month`|
|`com.appboy.enums.SdkFlavor`|`com.braze.enums.SdkFlavor`|
|`com.appboy.enums.NotificationSubscriptionType`|`com.braze.enums.NotificationSubscriptionType`|

 ## Testing Plan
 - Ran test suite locally. All tests passed.
 - Compiled a local artifact `android-appboy-kit-release.aar` and impoted that locally. Everything works OK.

### CODE REVIEW

**Review by mmustafa-tse — APPROVED** [review]
LGTM! @markvdouw let me know how it looks on your end

File: build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ buildscript {
 }
 
 plugins {
-    id "org.sonarqube" version "3.5.0.2730"
+    id "org.sonarqube" version "4.0.0.2929"
```
Comment (markvdouw) [inline build.gradle:19]: Sonarqube version shouldnt be changed, specially a major bump up. Will messup with our CICD for now

**Review by markvdouw — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (techouse) [reply build.gradle:19]: That came from https://github.com/mparticle-integrations/mparticle-android-integration-appboy/pull/123/commits/e040f5351eba1c08a12f22a264501d089511c180 and I'm not sure why it even got in there. I'll revert it.

**Comment by techouse** [comment]
@mmustafa-tse & @markvdouw can you have another look at this please?

**Review by mmustafa-tse — APPROVED** [review]
LGTM! already tested this code. awaiting @markvdouw

**Review by markvdouw — APPROVED** [review]
LGTM

**Comment by markvdouw** [comment]
I will wait for test to pass, merge it and run will be deployed soon.

**Pull request merged**